data/api/client-server/receipts.yaml

```
@@ -88,5 +98,17 @@ paths:
           description: This request was rate-limited.
           schema:
             "$ref": "definitions/errors/rate_limited.yaml"
+        400:
+          description: |-
+            The `thread_id` is not a string, or is empty, or it is provided for
+            a receipt of type `m.fully_read`, or the `event_id` is not related
+            to the `thread_id`.
```

We can be a bit more relaxed on error code descriptions:
```suggestion
            The `thread_id` is invalid in some way. For example:
            * It is not a string.
            * It is empty.
            * It is provided for an incompatible receipt type.
            * The `event_id` is not related to the `thread_id`.
```